identity/did/identity.go

```diff
@@ -129,8 +132,28 @@ func logTxHash(tx *types.Transaction) {
 	log.Infof("Transfer pending: 0x%x\n", tx.Hash())
 }
 
-func (i identity) AddKey(key Key) (chan *ethereum.WatchTransaction, error) {
-	contract, opts, err := i.prepareTransaction(*i.did)
+func (i identity) getDID(ctx context.Context) (did DID, err error) {
+	tc, err := contextutil.Account(ctx)
+	if err != nil {
+		return did, err
+	}
+
+	addressByte, err := tc.GetIdentityID()
+	if err != nil {
+		return did, err
+	}
+	did = NewDID(common.BytesToAddress(addressByte))
+	return did, nil
+
+}
+
+func (i identity) AddKey(ctx context.Context, key Key) (chan *ethereum.WatchTransaction, error) {
```

Comment:
add comments pls

Reply:
I thought we only add comments as required only to the interface method and not to the non-public implementation.

Reply:
adding comments to interface explains what method should do or return but adding it to the implementation explains implementation specific details.